pkg/comp-functions/functions/vshn-postgres-func/mailgun_alerting.go

```diff
@@ -0,0 +1,133 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	alertmanagerv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	runtime "github.com/vshn/appcat/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+func MailgunAlerting(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("Starting mailgn-alerting function")
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := iof.Observed.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+	// Wait for the next reconciliation in case instance namespace is missing
+	if comp.Status.InstanceNamespace == "" {
+		return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
+	}
+
+	if comp.Spec.Parameters.Monitoring.Email != "" {
+		log.Info("email not null")
+		err := deployAlertmanagerConfig(ctx, comp, iof)
+		if err != nil {
+			fmt.Println(err)
+			return runtime.NewFatalErr(ctx, "Can't add "+comp.Name+"-alertmanagerconfig-mailgun", err)
+		}
+	}
+
+	log.Info("Finishing mailgun-alerting function with NewNormal")
+
+	return runtime.NewNormal()
+}
+
+func deployAlertmanagerConfig(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, iof *runtime.Runtime) error {
+	log := controllerruntime.LoggerFrom(ctx)
+	var alertManagerConfigName = comp.Name + "-alertmanagerconfig-mailgun"
+	var alertManagerConfigSecretName = comp.Name + "-alertmanagerconfig-mailgun-secret"
+	receiverName := "mailgun"
+
+	ac := &alertmanagerv1alpha1.AlertmanagerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      alertManagerConfigName,
+			Namespace: comp.Status.InstanceNamespace,
+			Labels: map[string]string{
+				"alertManagerConfig": "mailgun",
```

Comment:
```suggestion
				"alert-manager-config": receiverName,
```
We should probably not use camel case in labels